src/events/mod.rs

```diff
@@ -387,7 +387,59 @@ impl<'a> BytesDecl<'a> {
         BytesDecl { element: start }
     }
 
-    /// Gets xml version, including quotes (' or ")
+    /// Gets xml version, excluding quotes (' or ")
+    ///
+    /// According to the [grammar], the version *must* be the first thing in the declaration.
+    /// This method tries to extract the first thing in the declaration and return it.
+    /// In case of multiple attributes value of the first one is returned.
+    ///
+    /// If version is missed in the declaration, or the first thing is not a version,
+    /// [`Error::XmlDeclWithoutVersion`] will be returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    /// use quick_xml::Error;
+    /// use quick_xml::events::{BytesDecl, BytesStart};
+    ///
+    /// // <?xml version='1.1'?>
+    /// let decl = BytesDecl::from_start(BytesStart::borrowed(b" version='1.1'", 0));
```

Comment:
For the sake of documenting the most convenient / foolproof method, maybe use [`BytesDecl::new`](https://docs.rs/quick-xml/0.22.0/quick_xml/events/struct.BytesDecl.html#method.new) instead

Reply:
Yes, that of course idiomatic method, but this three examples shows, how input transformed into `BytesDecl`. Because other tho examples shows, how semantically incorrect input is parsed, so as not to interrupt continuity the same methods is called in all three situations. It is possible, however, mention the right method

Reply:
Besides, I think that [BytesDecl::new](https://docs.rs/quick-xml/0.22.0/quick_xml/events/struct.BytesDecl.html#method.new) are the thing that should be changed. That too unsafe for public API, because it can generate invalid XML. I think, it will be better to change it to

```rust
pub enum Version {
  V1_0,
  V1_1,
  Other(String),
}

pub fn new(
  version: Version,
  // Option<Encoding> would be better, but this will require "encoding" feature
  encoding: Option<String>,
  standalone: Option<bool>,
);
```
instead